scripts/renderers.py

```diff
@@ -15,35 +15,85 @@
 
-
 class CsvRenderer():
     """
     Used to output toggles+annotations data as CSS
     """
 
-    def __init__(self):
-        pass
+    def transform_toggle_data_for_csv(self, envs_data):
+        """
+        Retrieve list of individual toggle datums from envs_data
 
-    def render_flag_csv_report(self, idas):
-        flag_toggles = []
-        header = set()
-        for ida_name, ida in idas.items():
-            for flag_toggle in ida.toggles['WaffleFlag']:
-                data_dict = flag_toggle.full_data()
-                header = header.union(set(data_dict.keys()))
-                flag_toggles.append(data_dict)
-        self.write_csv("test_flag.csv", flag_toggles, header)
-
-    def render_switch_csv_report(self, idas):
-        switch_toggles = []
-        header = set()
-        for ida_name, ida in idas.items():
-            for switch_toggle in ida.toggles['WaffleSwitch']:
-                data_dict = switch_toggle.full_data()
-                header = header.union(set(data_dict.keys()))
-                switch_toggles.append(data_dict)
-        self.write_csv("test_switch.csv", switch_toggles, header)
+        envs_data is a dict with a complex structure of environments, idas and toggles by toggle type.
+        Return a flattened list for each toggle in a specific environment in preparation for csv output.
+        """
+        toggles_data = []
+        for env, idas in envs_data.items():
+            for ida_name, ida in idas.items():
+                for toggle_type, toggles in ida.toggles.items():
+                    for toggle in toggles:
+                        data_dict = toggle.full_data()
+                        data_dict["toggle_type"] = toggle_type
+                        data_dict["ida_name"] = ida_name
+                        data_dict["env_name"] = env
+                        toggles_data.append(data_dict)
+        return toggles_data
+
+    def filter_and_sort_toggles(self, toggles_data, toggle_type_filter=None):
+        """
+        Arguments:
+            - toggles_data: list[dict] dicts should have all relevant data relating to one toggle
+            - toggle_type_filter: list of type names: there are multiple toggle types, so which would you like to output
+                     if set to None, everything will be outputed
+        Returns:
+            - list: sorted list of filtered toggle data.
+        """
 
+        # filter toggles by toggle_types
+        data_to_render = []
+        if not toggle_types:
+            data_to_render = toggles_data
+        else:
+            if isinstance(toggle_types, str):
+                toggle_types = [toggle_types]
+            for toggle_dict in toggles_data:
+                if toggle_dict["toggle_type"] in toggle_types:
+                    data_to_render.append(toggle_dict)
+
+        # sort data by either annotation_name or state_name
+        sorting_key = lambda datum: (datum.get("annotation_name", ""), datum.get("state_name", ""))
```

Comment:
Here and above, I thought we discussed having to combine into a single name?  Is that coming in the follow-up PR?

Reply:
Yep, this will be included in follow-up PR